client/state/selectors/is-calypsoify-gutenberg-enabled.js

```diff
@@ -1,53 +1,48 @@
 /** @format */
 
-/**
- * External dependencies
- */
-import { get } from 'lodash';
-
 /**
  * Internal dependencies
  */
 import { isEnabled } from 'config';
 import isVipSite from 'state/selectors/is-vip-site';
-import { isJetpackSite } from 'state/sites/selectors';
+import { getSiteAdminUrl, isJetpackMinimumVersion, isJetpackSite } from 'state/sites/selectors';
 import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
-import getSiteOptions from 'state/selectors/get-site-options';
 import getWordPressVersion from 'state/selectors/get-wordpress-version';
 import versionCompare from 'lib/version-compare';
 import isPluginActive from 'state/selectors/is-plugin-active';
+import { isHttps } from 'lib/url';
 
 export const isCalypsoifyGutenbergEnabled = ( state, siteId ) => {
 	if ( ! siteId ) {
 		return false;
 	}
 
-	if ( isEnabled( 'jetpack/gutenframe' ) ) {
-		if (
-			versionCompare(
-				get( getSiteOptions( state, siteId ), 'jetpack_version', 0 ),
-				'7.3-alpha',
-				'>='
-			)
-		) {
+	// We might want Calypsoify flows for Jetpack and Atomic sites.
+	if ( isJetpackSite( state, siteId ) || isSiteAutomatedTransfer( state, siteId ) ) {
+		// But only once they have been updated to WordPress version 5.0 or greater since it will provide Gutenberg
+		// editor by default.
+		const wpVersion = getWordPressVersion( state, siteId );
+		if ( versionCompare( wpVersion, '5.0', '<' ) ) {
 			return false;
 		}
-	}
 
-	// We do want Calypsoify flows for Atomic sites
-	if ( isSiteAutomatedTransfer( state, siteId ) ) {
-		const wpVersion = getWordPressVersion( state, siteId );
-
-		// But not if they activated Classic editor plugin (effectively opting out of Gutenberg)
+		// But not if they activated the Classic Editor plugin (effectively opting out of Gutenberg).
 		if ( isPluginActive( state, siteId, 'classic-editor' ) ) {
 			return false;
 		}
 
-		// But only once they have been updated to WordPress version 5.0 or greater
-		// Since it will provide Gutenberg editor by default
-		if ( versionCompare( wpVersion, '5.0', '>=' ) ) {
-			return true;
+		// We do want Gutenframe flows for JP/AT sites that have been updated to Jetpack 7.3 or greater since it will
+		// provide a way to handle the frame nonces verification. But only if the site has a SSL cert since the browser
+		// cannot embed insecure content in a resource loaded over a secure HTTPS connection.
+		if (
+			isEnabled( 'jetpack/gutenframe' ) &&
+			isJetpackMinimumVersion( state, siteId, '7.3-alpha' ) &&
+			isHttps( getSiteAdminUrl( state, siteId ) )
```

Comment:
This alone won't handle the dev environment properly; we _should_ allow dev (`calypso.localhost`) to load HTTP since neither has SSL, and are therefore compatible. Really, we want "if the parent is secure and the iframe is not" as our check. This can be a low-priority follow-up, though.

Reply:
See: https://github.com/orgs/Automattic/projects/88#card-20920479

Reply:
I'm not sure about this. On one hand, it will come handy when developing locally, but on the other hand we will have an environment that won't match the production behavior and that can leads to more time figuring out why something fails on production but not locally.

Reply:
It does match production behaviour though: the iframing is allowed if the result would not be mixed-content errors. I'm trying to imagine how else we can even test certain things without it.

Reply:
Solved by https://github.com/Automattic/wp-calypso/pull/32678